enos/modules/vault_verify_version/templates/verify-cluster-version.sh

```diff
@@ -13,25 +18,37 @@ fail() {
 
 test -x "$binpath" || fail "unable to locate vault binary at $binpath"
 
-binary_version_full=$($binpath version)
-# Get the Vault build tag
-binary_version=$(cut -d ' ' -f2 <<< $binary_version_full)
-# Strip the leading v
-semantic=$${binary_version:1}
-# Get the build timestamp
-build_date=$(cut -d ' ' -f5 <<< $binary_version_full)
-
 export VAULT_ADDR='http://127.0.0.1:8200'
+export VAULT_TOKEN='${vault_token}'
+
+if [[ ! -z "$builddate" ]]; then
+  build_date=$builddate
+else
+  build_date=$("$binpath" status -format=json | jq -Mr .build_date)
+fi
+
+if [[ "$(echo $version |awk -F'.' '{print $2}')" -ge 11 ]]; then
+  version_expected="Vault v$release ($sha), built $build_date"
+else
+  version_expected="Vault v$release ($sha)"
+fi
 
-# Ensure that the cluster version and build time match the binary installed
-vault_status=$("$binpath" status -format json)
-result=$(jq -Mr \
-  --arg version "$semantic" \
-  --arg build_date "$build_date" \
-  'select(.version == $version) | .build_date == $build_date' \
-  <<< $vault_status
-)
-
-if [[ "$result" != "true" ]]; then
-  fail "expected version $binary_version with build_date $build_date, got status $vault_status"
+case "$release" in
+  *+oss) [[ "$(echo $version |awk -F'.' '{print $2}')" -ge 11 ]] && \
+  version_expected="Vault v$version ($sha), built $build_date" || \
+  version_expected="Vault v$version ($sha)" ;;
+	*+ent) ;;
+	*+ent.hsm) version_expected="$version_expected (cgo)";;
+	*+ent.fips1402) version_expected="$version_expected (cgo)" ;;
+	*+ent.hsm.fips1402) version_expected="$version_expected (cgo)" ;;
+  *) fail "($release) file doesn't match any known license types"
+esac
+
+version_expected_nosha=$(echo "$version_expected" | awk '!($3="")' | sed 's/  / /' | sed -e 's/[[:space:]]*$//')
+version_output=$("$binpath" version)
+
+if [[ "$version_output" == "$version_expected_nosha" ]] || [[ "$version_output" == "$version_expected" ]]; then
+	echo "Version verification succeeded!"
```

Comment:
nit: inconsistent indentation in this script

Reply:
updated